rust/lancedb/src/remote/table.rs

```diff
@@ -149,82 +173,199 @@ impl<S: HttpSend> TableInternal for RemoteTable<S> {
     }
     async fn add(
         &self,
-        _add: AddDataBuilder<NoData>,
-        _data: Box<dyn RecordBatchReader + Send>,
+        add: AddDataBuilder<NoData>,
+        data: Box<dyn RecordBatchReader + Send>,
     ) -> Result<()> {
-        todo!()
+        let body = Self::reader_as_body(data)?;
+        let mut request = self
+            .client
+            .post(&format!("/table/{}/insert/", self.name))
+            .header(CONTENT_TYPE, ARROW_STREAM_CONTENT_TYPE)
+            .body(body);
+
+        match add.mode {
+            AddDataMode::Append => {}
+            AddDataMode::Overwrite => {
+                request = request.query(&[("mode", "overwrite")]);
+            }
+        }
+
+        let response = self.client.send(request).await?;
+
+        self.check_table_response(response).await?;
+
+        Ok(())
     }
     async fn build_plan(
         &self,
         _ds_ref: &DatasetReadGuard,
         _query: &VectorQuery,
         _options: Option<QueryExecutionOptions>,
     ) -> Result<Scanner> {
-        todo!()
+        Err(Error::NotSupported {
+            message: "build_plan is not supported on LanceDB cloud.".into(),
+        })
     }
     async fn create_plan(
         &self,
         _query: &VectorQuery,
         _options: QueryExecutionOptions,
     ) -> Result<Arc<dyn ExecutionPlan>> {
-        unimplemented!()
+        Err(Error::NotSupported {
+            message: "create_plan is not supported on LanceDB cloud.".into(),
+        })
     }
     async fn explain_plan(&self, _query: &VectorQuery, _verbose: bool) -> Result<String> {
-        todo!()
+        Err(Error::NotSupported {
+            message: "explain_plan is not supported on LanceDB cloud.".into(),
+        })
     }
     async fn plain_query(
         &self,
         _query: &Query,
         _options: QueryExecutionOptions,
     ) -> Result<DatasetRecordBatchStream> {
-        todo!()
+        Err(Error::NotSupported {
+            message: "plain_query is not yet supported on LanceDB cloud.".into(),
```

Comment:
"not yet" for this one.  You think it is more likely to be added at some point than the others?

Reply:
Well not necessarily. I'm going to fix this up in my follow up PR when I implement query.